### PR TITLE
LCAM-994|Remove transaction id

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-994)

Removed laa-transaction-id usage and deleted the redundant Constants file.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java [hunk 1]
```diff
@@ -52,27 +49,26 @@ public ApiPersistHardshipResponse persistHardship(ApiPersistHardshipRequest requ
                     new ParameterizedTypeReference<>() {
                     },
                     configuration.getMaatApi().getHardshipEndpoints().getPersistHardshipUrl(),
-                    Map.of(Constants.LAA_TRANSACTION_ID, laaTransactionId)
+                    Map.of()
```
Comment (nigelpain) [inline crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java:52]: Do we still need to provide an empty map here, or can we just not include when making the request via the API Client library?

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java [hunk 2]
```diff
@@ -77,12 +76,12 @@ public ResponseEntity<ApiCalculateHardshipByDetailResponse> calculateHardshipFor
     public ResponseEntity<ApiFindHardshipResponse> find(
             @PathVariable int hardshipReviewId,
             @Parameter(description = "Used to trace calls between services")
-            @RequestHeader(value = "Laa-Transaction-Id", required = false) String laaTransactionId) {
+            @RequestHeader(value = Constants.LAA_TRANSACTION_ID, required = false) String laaTransactionId) {
```
Comment (nigelpain) [inline crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java:79]: If were not passing through the transaction id during calls now due to the newly implemented tracing do we still need it defined as a header value here?

Comment (skasthuri36326) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java:79]: Logging the 'Laa-Transaction-Id' is required only in the entry point of MAAT Application to the Modernised Services. E.g. MAAT Application calling the Orchestration Service which is the entry point. Micrometer automatically adds the the traceId and spanId for the subsequent call logs. "Laa-Transaction-Id" can be removed from all the other service calls.

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java (hunk 1 above)
Comment (venkat980) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java:52]: The API client library is expecting a value to be passed, although passing null is resulting in the tests failure due to stubbing. Hence we are passing an empty map indicating there are no headers.

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java (hunk 2 above)
Comment (venkat980) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/controller/HardshipController.java]: Resolved.

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java (hunk 1 above)
Comment (skasthuri36326) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java:52]: @venkat980 Header is optional. We can use the below implementation in RestAPIClient?

```
public <R> R get(ParameterizedTypeReference<R> typeReference, String url, Object... urlVariables) {
        return getApiResponse(null, typeReference, url, null, HttpMethod.GET, null, urlVariables);
    }
```

Comment (venkat980) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/MaatCourtDataService.java:52]: Sri, currently we dont have this implementation for post.

**Review by skasthuri36326 — APPROVED** [review]

**Pull request merged**